pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +145,333 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
+	}
+
+	watcherIPlist := []string{}
+	for _, watcher := range rbdStatusObj.Watchers {
+		// split with separation '/' to remove nounce and concatenating `/32` to define a network with only one IP address
+		address := strings.Split(watcher.Address, "/")[0] + "/32"
+		watcherIPlist = append(watcherIPlist, address)
+	}
+	return watcherIPlist, nil
+}
+
+// func (c *clientCluster) fennceCephFSVolumes(
+// 	ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster,
+// 	clusterInfo *cephclient.ClusterInfo, cephFSPV corev1.PersistentVolume) error {
+
+// 	logger.Infof("cephfs PV NAME %v", cephFSPV.Spec.CSI.VolumeAttributes)
+
+// 	status, err := cephclient.StatusWithUser(c.context, clusterInfo)
+// 	if err != nil {
+// 		return fmt.Errorf("failed to get ceph status for check active mds. %v", err)
+// 	}
+
+// 	var activeMDS string
+// 	for _, fsRank := range status.Fsmap.ByRank {
+// 		if fsRank.Status == "up:active" {
+// 			activeMDS = fsRank.Name
+// 		}
+// 	}
+
+// 	args := []string{"tell", fmt.Sprintf("mds.%s", activeMDS), "client", "ls", "--format", "json"}
+// 	cmd := cephclient.NewCephCommand(c.context, clusterInfo, args)
+// 	cmd.JsonOutput = true
+
+// 	buf, err := cmd.Run()
+// 	if err != nil {
+// 		return fmt.Errorf("failed to list watchers for cephfs pool/subvoumeName %s/%s. %v", cephFSPV.Spec.CSI.VolumeAttributes["pool"], cephFSPV.Spec.CSI.VolumeAttributes["subvolumeName"], err)
+// 	}
+// 	ips, err := cephFSMDSClientMarshal(buf, cephFSPV)
+// 	if err != nil || ips == nil {
+// 		return fmt.Errorf("failed to unmarshal cephfs mds  output. %v", err)
+// 	}
+
+// 	err = c.createNetworkFence(ctx, cephFSPV, node, cluster, ips)
+// 	if err != nil {
+// 		return fmt.Errorf("failed to create network fence for node %q. %v", node.Name, err)
+// 	}
+
+// 	return nil
+// }
+
+// func cephFSMDSClientMarshal(output []byte, cephFSPV corev1.PersistentVolume) ([]string, error) {
+// 	type entity struct {
+// 		Addr struct {
+// 			Addr  string `json:"addr"`
+// 			Nonce int    `json:"nonce"`
+// 		} `json:"addr"`
+// 	}
+
+// 	type clientMetadata struct {
+// 		Root string `json:"root"`
+// 	}
+
+// 	type cephFSData struct {
+// 		Entity         entity         `json:"entity"`
+// 		ClientMetadata clientMetadata `json:"client_metadata"`
+// 	}
+
+// 	var data []cephFSData
+// 	err := json.Unmarshal(output, &data)
+// 	if err != nil {
+// 		return []string{}, err
+// 	}
+
+// 	for _, d := range data {
+// 		if cephFSPV.Spec.CSI.VolumeAttributes["subvolumePath"] == d.ClientMetadata.Root {
+// 			logger.Infof("cephfs mds client ips to fence %v", d.Entity.Addr)
+// 			return []string{d.Entity.Addr.Addr + "/32"}, nil
+// 		}
+// 	}
+
+// 	return []string{}, nil
+// }
+
+func (c *clientCluster) createNetworkFence(ctx context.Context, pv corev1.PersistentVolume, node *corev1.Node, cluster *cephv1.CephCluster, cidr []string) error {
+	logger.Warningf("Blocking node IP %s", cidr)
```

Comment:
If there are multiple rbd volumes on the same node, will we see this same message for every volume? Can we change it to only print once?

Reply:
no, we are fencing per node level not the volume/pv level.

Reply:
`createNetworkFence()` is called by `fenceRbdImage()`, which is called by a `for` loop, so it looks like the node would be fenced multiple times.

Reply:
actually, missed adding break if err !=nil. So if `fenceRbdImage` is success it will come out